Mathlib/FieldTheory/LinearDisjoint.lean

```diff
@@ -0,0 +1,361 @@
+
+-/
+
+open scoped Classical TensorProduct
+
+open Module IntermediateField
+
+noncomputable section
+
+universe u v w
+
+namespace IntermediateField
+
+variable {F : Type u} {E : Type v} [Field F] [Field E] [Algebra F E]
+
+variable (A B : IntermediateField F E)
+
+variable (L : Type w) [Field L] [Algebra F L] [Algebra L E] [IsScalarTower F L E]
+
+/-- If `A` is an intermediate field of `E / F`, and `E / L / F` is a field extension tower,
+then `A` and `L` are linearly disjoint, if they are linearly disjoint as subalgebras of `E`
+(`Subalgebra.LinearDisjoint`). -/
+protected abbrev LinearDisjoint : Prop :=
+  A.toSubalgebra.LinearDisjoint (IsScalarTower.toAlgHom F L E).range
+
+theorem linearDisjoint_iff :
+    A.LinearDisjoint L ↔ A.toSubalgebra.LinearDisjoint (IsScalarTower.toAlgHom F L E).range :=
+  Iff.rfl
+
+variable {A B L}
+
+/-- Two intermediate fields are linearly disjoint if and only if
+they are linearly disjoint as subalgebras. -/
+theorem linearDisjoint_iff' :
+    A.LinearDisjoint B ↔ A.toSubalgebra.LinearDisjoint B.toSubalgebra := by
+  rw [linearDisjoint_iff]
+  congr!
+  ext; simp
+
+/-- Linearly disjoint is symmetric. -/
+theorem LinearDisjoint.symm (H : A.LinearDisjoint B) : B.LinearDisjoint A :=
+  linearDisjoint_iff'.2 (linearDisjoint_iff'.1 H).symm
+
+/-- Linearly disjoint is symmetric. -/
+theorem linearDisjoint_symm : A.LinearDisjoint B ↔ B.LinearDisjoint A :=
+  ⟨LinearDisjoint.symm, LinearDisjoint.symm⟩
+
+namespace LinearDisjoint
+
+variable (A) in
+theorem self_right : A.LinearDisjoint F :=
+  Subalgebra.LinearDisjoint.bot_right _
+
+variable (A) in
+theorem bot_right : A.LinearDisjoint (⊥ : IntermediateField F E) :=
+  linearDisjoint_iff'.2 (Subalgebra.LinearDisjoint.bot_right _)
+
+variable (F E L) in
+theorem bot_left : (⊥ : IntermediateField F E).LinearDisjoint L :=
+  Subalgebra.LinearDisjoint.bot_left _
+
+/-- If `A` and `L` are linearly disjoint, then any `F`-linearly independent family on `A` remains
+linearly independent over `L`. -/
+theorem linearIndependent_left (H : A.LinearDisjoint L)
+    {ι : Type*} {a : ι → A} (ha : LinearIndependent F a) :
+    LinearIndependent L (A.val ∘ a) :=
+  (Subalgebra.LinearDisjoint.linearIndependent_left_of_flat H ha).map_of_injective_injective
+    (AlgEquiv.ofInjectiveField (IsScalarTower.toAlgHom F L E)) (AddMonoidHom.id E)
+    (by simp) (by simp) (fun _ _ ↦ by simp_rw [Algebra.smul_def]; rfl)
+
+/-- If there exists an `F`-basis of `A` which remains linearly independent over `L`, then
+`A` and `L` are linearly disjoint. -/
+theorem of_basis_left {ι : Type*} (a : Basis ι F A)
+    (H : LinearIndependent L (A.val ∘ a)) :
+    A.LinearDisjoint L := by
+  rw [linearDisjoint_iff]
+  refine Subalgebra.LinearDisjoint.of_basis_left _ _ a ?_
+  exact H.map_of_surjective_injective
+    (AlgEquiv.ofInjectiveField (IsScalarTower.toAlgHom F L E)) (AddMonoidHom.id E)
+    (AlgEquiv.surjective _) (by simp) (fun _ _ ↦ by simp_rw [Algebra.smul_def]; rfl)
```

Comment:
looks like pure term-mode is possible here
(also for `of_basis_right'`, `of_basis_mul'`, `of_finrank_sup`, maybe more)